### ISSUE TITLE
Incorrect autocomplete data typings

### ISSUE THREAD

**Issue by gc** [discordjs/discord-api-types#230]
**Please describe the problem you are having in as much detail as possible:**

It seems like the Autocomplete types are incorrect/incomplete, if it is indeed incorrect, I'm willing to make a PR, if nobody else does.

**Include a reproducible code sample here, if possible:**

```ts
import { APIInteraction, InteractionType } from 'discord-api-types/v9';

const interaction: APIInteraction = {} as APIInteraction;
if (interaction.type === InteractionType.ApplicationCommandAutocomplete) {
    // Property 'options' does not exist on type 'APIApplicationCommandInteractionData'.
    //  Property 'options' does not exist on type 'APIUserApplicationCommandInteractionData'.
    interaction.data?.options;
}
```

I expect i should be able to do this:
```ts
interaction.data?.options[0]?.value // string | number | undefined
```

**Further details:**

NodeJS Version: v16.9.1
discord-api-types Version: 0.25.0-next.bc1d03e.1635987681

**Comment by vladfrangu** [comment]
CC @suneettipirneni

**Comment by suneettipirneni** [comment]
will be closed in #234
